server/utils/csv/csvProcess.js

```diff
@@ -0,0 +1,53 @@
+const mongoose = require("mongoose");
+const multer = require('multer');
+const csv = require('csv-parser');
+const fs = require('fs');
+
+const { validateProspectInput } = require("../../validation/prospect");
+const Prospect = require("../../models/Prospect");
+
+async function processCsvData(data) {
+    let skippedCount = 0;
+    let successCount = 0;
+    let uploadArray = [];
+    for(let i = 0; i < data.length; i++) {
```

Comment:
Instead of doing a loop and then searching for each Prospect after each other, there are two improvements you could do:
1. use Promises.all 
2. Better than that, is to make one call to MongoDB using the in query (https://docs.mongodb.com/manual/reference/operator/query/in/). You could also set up a projection just to return the emails that have been found, and then you can put those emails in a set, and find the difference between data (as a set as well) and the found set.

Reply:
I did a pretty significant refactor of the method based on your comments. Looks like it cut down time quite a bit. Will be merging to dev now